libs/core/src/lib/avatar-group/avatar-group.component.ts

```diff
@@ -0,0 +1,182 @@
+
+export type AvatarGroupType = 'group' | 'individual';
+export type AvatarGroupOverflowButtonColor = 'neutral' | 'random' | ColorAccent;
+
+let avatarGroupUniqueId = 0;
+
+@Component({
+    selector: 'fd-avatar-group',
+    templateUrl: './avatar-group.component.html',
+    styleUrls: ['./avatar-group.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.Default
+})
+export class AvatarGroupComponent implements OnChanges, OnInit, AfterViewInit, OnDestroy {
+    /** Id of the Avatar Group. */
+    @Input()
+    @HostBinding('attr.id')
+    id = `fd-avatar-group-${avatarGroupUniqueId++}`;
+
+    /** The size of the Avatar Group. Options include: *xs*, *s*, *m*, *l* and *xl* (default: *s*). */
+    @Input()
+    size: Size = 's';
+
+    /** The type of the Avatar Group. Options include: *group* and *individual* (default: *group*). */
+    @Input()
+    type: AvatarGroupType = 'group';
+
+    /** Aria-label for Avatar Group. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Counter for all avatars. */
+    allItemsCount = 0;
+
+    /** Counter for visible in main content avatars. */
+    visibleItemsCount = 0;
+
+    /** Counter for visible in overflow popover avatars. */
+    overflowItemsCount = 0;
+
+    /** @hidden Avatar Group items. */
+    @ContentChildren(forwardRef(() => AvatarGroupItemDirective), { descendants: true })
+    mainItems: QueryList<AvatarGroupItemDirective>;
+
+    /** @hidden */
+    @ViewChild('avatarGroupContainer')
+    avatarGroupContainer: ElementRef;
+
+    /** @hidden */
+    rootClassNames: Record<string, boolean | undefined | null>;
+
+    /** @hidden */
+    get isGroupType(): boolean {
+        return this.type === 'group';
+    }
+
+    /** @hidden */
+    private get _avatarGroupWidth(): number {
+        return (this.avatarGroupContainer?.nativeElement as HTMLElement)?.offsetWidth;
+    }
+
+    /** @hidden */
+    private get _avatarGroupItemWidth(): number {
+        return (this.mainItems?.first?.elementRef?.nativeElement as HTMLElement)?.offsetWidth;
+    }
+
+    /** @hidden */
+    private get _avatarGroupItemWithMarginsWidth(): number {
+        const elementStyles = getComputedStyle(this.mainItems?.first?.elementRef?.nativeElement);
+        return this._avatarGroupItemWidth
+            + parseInt(elementStyles.marginLeft, 10)
+            + parseInt(elementStyles.marginRight, 10);
+    }
+
+    /** @hidden */
+    private _subscription = new Subscription();
+
+    /** @hidden */
+    ngOnInit(): void {
+        const sub = fromEvent(window, 'resize')
+            .pipe(distinctUntilChanged())
+            .subscribe(_ => this._onResize());
+
+        this._subscription.add(sub);
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this._assignCssClasses();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._subscription.add(of(true).pipe(delay(5)).subscribe(() => this._collapseItems()));
```

Comment:
To prevent `ExpressionChangedAfterItHasBeenCheckedError` we need some delay. It could be `setTimeout` as well, but I'm not sure which way better _(rxjs delay or setTimeout)_